resources/qml/MachineAction.qml

```diff
@@ -0,0 +1,15 @@
+import QtQuick 2.2
+
+Item
+{
+    id: contentItem
+    // Connect the finished property change to completed signal.
+    property var finished: manager.finished
+    onFinishedChanged: if(manager.finished) {completed()}
+    signal completed()
+
+    function reset()
+    {
+        manager.reset()
+    }
+}
```

Comment:
It [c|w]ould be better to include the pageTitle (and pageDescription) that is in each MachineAction here, much like how PreferencePage.qml does it, to enforce some uniformity.